synapse/storage/events.py

```diff
@@ -220,13 +221,35 @@ class EventsStore(
     EventsWorkerStore,
     BackgroundUpdateStore,
 ):
-
     def __init__(self, db_conn, hs):
         super(EventsStore, self).__init__(db_conn, hs)
 
         self._event_persist_queue = _EventPeristenceQueue()
         self._state_resolution_handler = hs.get_state_resolution_handler()
 
+        # Collect metrics on the number of forward extremities that exist.
+        self._current_forward_extremities_amount = {}
+
+        BucketCollector(
+            "synapse_forward_extremities",
+            lambda: self._current_forward_extremities_amount,
```

Comment:
You'll want to configure the buckets a bit here, something like `[1, 2, 3, 5, 7, 10, 15, 20, 50, 100, 200, 500]`, I think the default buckets are a lot larger.

Reply:
done